File: src/main/java/com/grepp/curdsample/dto/TaskDescription.java
package com.grepp.curdsample.dto;

import com.grepp.curdsample.domain.Task;
import com.grepp.curdsample.util.PriorityResolver;
import com.grepp.curdsample.util.TimeFormatter;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TaskDescription {

    private String code;

    private String title;
    private String description;

    private Integer priority;
    private boolean completeStatus;

    private String startDate;
    private String endDate;

    private String createdAt;
    private String updatedAt;

    public static TaskDescription from(Task task) {
        return new TaskDescription(
                task.getCode(),
                task.getTitle(),
                task.getDescription(),
                task.getPriority(),
                task.isCompleteStatus(),
                TimeFormatter.convertToStr(task.getStartTime()),
                TimeFormatter.convertToStr(task.getEndTime()),
                TimeFormatter.convertToStr(task.getCreatedAt()),
                TimeFormatter.convertToStr(task.getUpdatedAt())
        );
    }

    public String getPriorityLevel() {
        return PriorityResolver.resolve(priority);
    }

}

File: src/main/java/com/est/curdsample/dto/TaskDescription.kt
package com.est.curdsample.dto

import com.est.curdsample.domain.Task
import com.est.curdsample.util.convertToStr
import com.est.curdsample.util.priorityResolve

data class TaskDescription(
    val code: String,
    val title: String,
    val description: String,
    val priority: Int,
    val completeStatus: Boolean,
    val startDate: String,
    val dueDate: String,
    val createdAt: String,
    val updatedAt: String
) {

    val priorityLevel: String
        get() = priorityResolve(priority)

    companion object {
        @JvmStatic
        fun from(task: Task): TaskDescription {
            return TaskDescription(
                task.code,
                task.title,
                task.description,
                task.priority,
                task.completeStatus,
                convertToStr(task.startTime),
                convertToStr(task.endTime),
                convertToStr(task.createdAt),
                convertToStr(task.updatedAt)
            )
        }
    }
}
